src/app/member/[memberIdentifier]/study/[studyIdentifier]/review/page.tsx

```diff
@@ -8,15 +10,19 @@ import { ReviewControls } from '@/app/member/[memberIdentifier]/study/[studyIden
 import { StudyJobFiles } from '@/app/member/[memberIdentifier]/study/[studyIdentifier]/review/study-job-files'
 import { StudyResults } from '@/app/member/[memberIdentifier]/study/[studyIdentifier]/review/study-results'
 import { first } from 'remeda'
-
-export const dynamicParams = true
+import { auth } from '@clerk/nextjs/server'
+import { getMemberUserFingerprint } from '@/app/account/keys/user-key-actions'
 
 export default async function StudyReviewPage(props: {
     params: Promise<{
         memberIdentifier: string
         studyIdentifier: string
     }>
 }) {
+    const { userId: clerkId } = await auth()
+    if (!clerkId) return null
+    const fingerprint = await getMemberUserFingerprint(clerkId)
```

Comment:
```suggestion
    const fingerprint = (clerkId && await getMemberUserFingerprint(clerkId)) || null
```

Dunno if it's useful but if the user isn't signed in then they can still see the results but not be able to decrypt the results. This way there isn't a 3rd UI (empty page) that we need to consider